components/net/fetch/methods.rs

```diff
@@ -277,6 +277,19 @@ pub fn main_fetch(request: Rc<Request>,
 
         // Step 16
         // TODO this step (CSP/blocking)
+        if request.type_ == Type::Script {
```

Comment:
Add a comment that this is handling the "should internalResponse to request be blocked due to its MIME type" case, and add TODO comments for "should internalResponse to request be blocked as mixed content [MIX]", "should internalResponse to request be blocked by Content Security Policy [CSP]" and "should internalResponse to request be blocked due to nosniff"?